model/src/order.rs

```diff
@@ -159,6 +176,7 @@ pub struct OrderCreation {
     pub partially_fillable: bool,
     pub signature: Signature,
     pub signing_scheme: SigningScheme,
+    pub from: Option<H160>,
```

Comment:
I don't like that we will carry this field around to all the callsites that should really be using OrderMetadata.owner (from will be sometimes set which might make people believe they could be accessing it).

Maybe we can introduce a new type

```rust
struct OrderCreationPayload {
  #[serde(flatten)]
  pub order_creation: OrderCreation,
  pub from: Option<H160>,
}
```

This way the API would be like currently (flat with an optional from field) but all APIs that return orders would still only return OrderCreation without the From field (which can be read from Metadata)

Reply:
I fixed this up according to your suggestion. Looks a lot better now and require less overall changes.